src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -619,6 +626,61 @@ class NetworkOPsImp final
     std::vector <TransactionStatus> mTransactions;
 
     StateAccounting accounting_ {};
+
+private:
+    struct Stats
+    {
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , disconnected_duration (collector->make_gauge("State_Accounting","Disconnected_duration"))
+            , connected_duration (collector->make_gauge("State_Accounting","Connected_duration"))
+            , syncing_duration(collector->make_gauge("State_Accounting", "Syncing_duration"))
+            , tracking_duration (collector->make_gauge("State_Accounting", "Tracking_duration"))
+            , full_duration (collector-> make_gauge("State_Accounting", "Full_duration"))
+            , disconnected_trasitions (collector->make_gauge("State_Accounting","Disconnected_trasitions"))
+            , connected_trasitions (collector->make_gauge("State_Accounting","Connected_trasitions"))
+            , syncing_trasitions(collector->make_gauge("State_Accounting", "Syncing_trasitions"))
+            , tracking_trasitions (collector->make_gauge("State_Accounting", "Tracking_trasitions"))
+            , full_trasitions (collector-> make_gauge("State_Accounting", "Full_trasitions"))
+            { }
+
+        beast::insight::Hook hook;
+        beast::insight::Gauge disconnected_duration;
+        beast::insight::Gauge connected_duration;
+        beast::insight::Gauge syncing_duration;
+        beast::insight::Gauge tracking_duration;
+        beast::insight::Gauge full_duration;
+
+        beast::insight::Gauge disconnected_trasitions;
+        beast::insight::Gauge connected_trasitions;
+        beast::insight::Gauge syncing_trasitions;
+        beast::insight::Gauge tracking_trasitions;
+        beast::insight::Gauge full_trasitions;
+    };
+
+    Stats m_stats;
+
+private:
+    void collect_metrics()
+    {     
+        auto [counters, mode, start] = accounting_.getCounterData();
+
+        auto const current = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now() - start);
+        counters[static_cast<std::size_t>(mode)].dur += current;
```

Comment:
can you explain why you need to update the local copy of data in `counters` here? Since this data is being obtained from the `StateAccounting` instance, I would have expected its duration data to be up-to-date.

Reply:
The current mode's duration is not up to date. It has to be manually incremented when observed. Here's another spot where we do the same thing:

https://github.com/ripple/rippled/blob/90d9ca901d06b9da1653526c234bc4b00e27070b/src/ripple/app/misc/NetworkOPs.cpp#L3490

In this case, we are observing and updating the duration of the current mode before returning a JSON object.

Reply:
yes - but in your case, you are updating a local *copy* of the array (`getCounterData` returns a tuple that has a copy of the data). Is that what you want, or do you want to update the array stored in `accounting_` ?

Reply:
Yes, I want to update the local copy and not the copy in accounting_.

Reply:
I'm confused. Can you explain _why_?

Reply:
I want to update the local copy because it is stale when I get it (the duration of the _current_ state in `accounting_` is stale therefore the local copy is stale too). This is because the durations of each mode in `accounting_` are only updated when the state changes and a new mode is set see [here](https://github.com/ripple/rippled/blob/cc4cefaa4b42c82da4f249eb3f54a77d2bcdc2d6/src/ripple/app/misc/NetworkOPs.cpp#L3464).  I could have also updated the array stored in `accounting_`, but it wouldn't benefit us because other places in the code similarly make a local copy and update it to get the _most_ up-to-date duration. An example if this is [here](https://github.com/ripple/rippled/blob/cc4cefaa4b42c82da4f249eb3f54a77d2bcdc2d6/src/ripple/app/misc/NetworkOPs.cpp#L3488). If I updated the duration of the current state in the array in `accounting_`, it would immediately become stale and need to be updated when observed again anyways.